storage/client_status_test.go

```diff
@@ -0,0 +1,183 @@
+		t.Errorf("%v: actual Live Bytes is not greater or equal to expected\nexpected: %+v\nactual: %v\n", testNumber, expectedStoreStatus, storeStatus)
+	}
+	if storeStatus.Stats.KeyBytes < expectedStoreStatus.Stats.KeyBytes {
+		t.Errorf("%v: actual Key Bytes is not greater or equal to expected\nexpected: %+v\nactual: %v\n", testNumber, expectedStoreStatus, storeStatus)
+	}
+	if storeStatus.Stats.ValBytes < expectedStoreStatus.Stats.ValBytes {
+		t.Errorf("%v: actual Val Bytes is not greater or equal to expected\nexpected: %+v\nactual: %v\n", testNumber, expectedStoreStatus, storeStatus)
+	}
+	if storeStatus.Stats.LiveCount < expectedStoreStatus.Stats.LiveCount {
+		t.Errorf("%v: actual Live Count is not greater or equal to expected\nexpected: %+v\nactual: %v\n", testNumber, expectedStoreStatus, storeStatus)
+	}
+	if storeStatus.Stats.KeyCount < expectedStoreStatus.Stats.KeyCount {
+		t.Errorf("%v: actual Key Count is not greater or equal to expected\nexpected: %+v\nactual: %v\n", testNumber, expectedStoreStatus, storeStatus)
+	}
+	if storeStatus.Stats.ValCount < expectedStoreStatus.Stats.ValCount {
+		t.Errorf("%v: actual Val Count is not greater or equal to expected\nexpected: %+v\nactual: %v\n", testNumber, expectedStoreStatus, storeStatus)
+	}
+	return storeStatus
+}
+
+// TestStoreStatus checks the store status after each range scan to ensure that
+// it is being updated correctly.
+func TestStoreStatus(t *testing.T) {
+	defer leaktest.AfterTest(t)
+	ctx := &storage.TestStoreContext
+	ctx.ScanInterval = time.Duration(10 * time.Millisecond)
+	store, stopper := createTestStoreWithEngine(t, engine.NewInMem(proto.Attributes{}, 10<<20), hlc.NewClock(hlc.NewManualClock(0).UnixNano), true, ctx)
+	defer stopper.Stop()
+	splitKey := proto.Key("b")
+	content := proto.Key("test content")
+
+	expectedStoreStatus := &proto.StoreStatus{
+		StoreID:    store.Ident.StoreID,
+		NodeID:     1,
+		RangeCount: 1,
+		Stats: proto.MVCCStats{
+			LiveBytes: 1,
+			KeyBytes:  1,
+			ValBytes:  1,
+			LiveCount: 1,
+			KeyCount:  1,
+			ValCount:  1,
+		},
+	}
+	store.WaitForRangeScanCompletion()
+	oldstats := compareStoreStatus(t, store, expectedStoreStatus, 0)
+
+	// Write some values left and right of the proposed split key.
+	rng := store.LookupRange([]byte("a"), nil)
+	pArgs, pReply := putArgs([]byte("a"), content, rng.Desc().RaftID, store.StoreID())
+	if err := store.ExecuteCmd(pArgs, pReply); err != nil {
+		t.Fatal(err)
+	}
+	pArgs, pReply = putArgs([]byte("c"), content, rng.Desc().RaftID, store.StoreID())
+	if err := store.ExecuteCmd(pArgs, pReply); err != nil {
+		t.Fatal(err)
+	}
+
+	expectedStoreStatus = &proto.StoreStatus{
+		StoreID:    store.Ident.StoreID,
+		NodeID:     1,
+		RangeCount: 1,
+		Stats: proto.MVCCStats{
+			LiveBytes: 1,
+			KeyBytes:  1,
+			ValBytes:  1,
+			LiveCount: oldstats.Stats.LiveCount + 1,
+			KeyCount:  oldstats.Stats.KeyCount + 1,
+			ValCount:  oldstats.Stats.ValCount + 1,
+		},
+	}
+	store.WaitForRangeScanCompletion()
```

Comment:
Is it possible that the scan has already started by the time you executed the Put? (so you need to wait for a scan to *start* after the puts, and then wait for that scan to complete)